### PR TITLE
fix: Simplify macro statement expansion handling

### PR DESCRIPTION
**Pull request by Veykril**
I only meant to fix https://github.com/rust-lang/rust-analyzer/issues/12644 but that somehow turned into a rewrite of the statement handling ... at least this fixes a few more issues in the IDE layer now

### CODE REVIEW

File: crates/hir/src/semantics.rs [hunk 1]
```diff
@@ -1016,7 +1016,7 @@ impl<'db> SemanticsImpl<'db> {
             None => return false,
         };
         let macro_call = self.find_file(macro_call.syntax()).with_value(macro_call);
-        sa.is_unsafe_macro_call(self.db, macro_call)
+        dbg!(sa.is_unsafe_macro_call(self.db, macro_call))
```
Comment (lnicola) [inline crates/hir/src/semantics.rs:1019]: ^

File: crates/hir-ty/src/tests/macros.rs [hunk 2]
```diff
@@ -335,16 +338,16 @@ fn recurisve_macro_expanded_in_stmts() {
...
+            !0..13 'ng!{[leta=]3}': ()
+            !0..13 'ng!{[leta]=3}': ()
+            !0..13 'ng!{[let]a=3}': ()
             !3..4 'a': i32
             !5..6 '3': i32
             196..237 '{     ...= a; }': ()
-            229..230 'b': i32
-            233..234 'a': i32
+            229..230 'b': {unknown}
+            233..234 'a': {unknown}
```
Comment (Veykril) [inline crates/hir-ty/src/tests/macros.rs:350]: Unsure why this is happening now 😕 All the other tests changes seem correct to me, but this one obviously isn't

**Comment by lnicola** [comment]
This brought down our unknown types by a lot (around July 2):

image

**Comment by Veykril** [comment]
Oh nice, we do infer types on some more expression now that we didn't look at before so I suppose this makes sense

**Pull request merged**